### PR TITLE
mongodb-4_0: drop

### PR DESCRIPTION
**Pull request by wegank**
###### Description of changes

MongoDB 4.0 was EOL in April 2022, broken on trunk (#209175), depends on Python 2 (#201859), and (still!) only the UniFi Video module uses it.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by wegank** [comment]
@ofborg eval

**Review by gador — APPROVED** [review]
LGTM

**Review by fabianhjr — APPROVED** [review]
LGTM

**Review by SuperSandro2000 — APPROVED** [review]

**Comment by SuperSandro2000** [comment]
#209175 did the samem, so should be fine, right?

File: nixos/modules/services/video/unifi-video.nix [hunk 1]
```diff
@@ -123,7 +123,7 @@ in
 
     mongodbPackage = mkOption {
       type = types.package;
-      default = pkgs.mongodb-4_0;
+      default = pkgs.mongodb-4_2;
```
[the remark this answers is not in the document]
Comment (wegank) [reply nixos/modules/services/video/unifi-video.nix:126]: Updating from 4.0 to 4.2 is officially supported, so this should be fine.

**Review by thiagokokada — APPROVED** [review]
With this removal we can set `sconsPackages.scons_3_1_1` to use `python3` directly and remove the `scons_3_1_1.overrides { python = python3; }` inside `mongodb`.

But this can be done in another PR.

**Pull request merged**